back-end/sports_platform/sports_platform_api/views/user_views.py

```diff
@@ -1,3 +1,27 @@
-from django.shortcuts import render
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+from ..validation import user_validation
+from ..controllers.guest import Guest
+import re 
 
-# We will create endpoints here.
\ No newline at end of file
+@api_view(['POST'])
+def create_user(request):    
+    validation = user_validation.User(data=request.data)
+    if not validation.is_valid():
+        return Response(validation.errors, status=400)
+    if 'password' not in request.data:
+        return Response('Password must be provided', status=400)
+    if 'identifier' not in request.data:
+        return Response('Identifier must be provided', status=400)
+    password = request.data['password']
+    if not(re.search(r'^[a-zA-Z\._\*]*$', password) and (len(password)>=8) and (len(password)<=15)):
+        return Response(data='Password Requirments are not satisfied', status=400)
+
+    try:
+        guest= Guest(request.data['email'], password)
```

Comment:
request.data['email']-> for this I think we should get the email after validation using validation['email']. This is also valid on line 22, i think.

Reply:
I did not know that we can call `validation['email']`. What is the difference between this and  `request.data['email']`?

Reply:
For example maybe not for this case but if the input was '2' and ve needed an integer field request.data['number'] would be '2' but validation['number'] would be 2. So we should always use validated one I think.

Reply:
okay I see, then validation also modifies the input, right? I thought it only checks the input format is the desired or not.